src/D2L.Security.OAuth2/Keys/Default/MonoRsaSecurityKey.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace D2L.Security.OAuth2.Keys.Default {
+
+	internal sealed class MonoRsaSecurityKey : AsymmetricSecurityKey {

```

Comment:
```
+#if __MonoCS__ || MONO
 +			if( key is MonoRsaSecurityKey || key is RsaSecurityKey ) {
 +#else
 +			if ( key is RsaSecurityKey ) {
 +#endif
```

 You mean these type checks? I don't think they'll have to change at all if you were to add `#if __MonoCS || MONO` around the definition...?